workitem/link/link_revision_repository.go

```diff
@@ -0,0 +1,70 @@
+package link
+
+import (
+	"context"
+
+	"fmt"
+
+	"time"
+
+	"github.com/almighty/almighty-core/errors"
+	"github.com/almighty/almighty-core/log"
+	"github.com/jinzhu/gorm"
+	uuid "github.com/satori/go.uuid"
+)
+
+// RevisionRepository encapsulates storage & retrieval of historical versions of work item links
+type RevisionRepository interface {
+	// Create stores a new revision for the given work item link.
+	Create(ctx context.Context, modifierID uuid.UUID, revisionType RevisionType, l WorkItemLink) error
+	// List retrieves all revisions for a given work item link
+	List(ctx context.Context, workitemID uuid.UUID) ([]Revision, error)
+}
+
+// NewRevisionRepository creates a GormCommentRevisionRepository
+func NewRevisionRepository(db *gorm.DB) *GormWorkItemLinkRevisionRepository {
+	repository := &GormWorkItemLinkRevisionRepository{db}
+	return repository
+}
+
+// GormCommentRevisionRepository implements CommentRevisionRepository using gorm
+type GormWorkItemLinkRevisionRepository struct {
+	db *gorm.DB
+}
+
+// Create stores a new revision for the given work item link.
+func (r *GormWorkItemLinkRevisionRepository) Create(ctx context.Context, modifierID uuid.UUID, revisionType RevisionType, l WorkItemLink) error {
+	log.Info(nil, map[string]interface{}{
+		"pkg":               "workitem/link",
+		"modifier_identity": modifierID,
+		"revision_type":     revisionType,
+	}, "Storing a revision after operation on work item link.")
+	tx := r.db
+	revision := &Revision{
+		ModifierIdentity:     modifierID,
+		Time:                 time.Now(),
+		Type:                 revisionType,
+		WorkItemLinkID:       l.ID,
+		WorkItemLinkVersion:  l.Version,
+		WorkItemLinkSourceID: l.SourceID,
+		WorkItemLinkTargetID: l.TargetID,
+		WorkItemLinkTypeID:   l.LinkTypeID,
+	}
+	if err := tx.Create(&revision).Error; err != nil {
+		return errors.NewInternalError(fmt.Sprintf("Failed to create new work item link revision: %s", err.Error()))
```

Comment:
@xcoulon sure: https://github.com/golang/go/wiki/CodeReviewComments#error-strings or our https://github.com/almighty/almighty-core/blob/master/.github/CONTRIBUTING.md which refers to that page.